### ISSUE TITLE
Can't find run_pretraining.sh

### ISSUE THREAD

**Issue by DEDSEC-Roger** [habla-liaa/encodecmae#2]
Hi! I can't find the "scripts" folder, hence I can't find "run_pretraining.sh" file. How should I pretrain the model?

**Comment by mrpep** [comment]
Hi! The scripts are available for the previous version of the paper: https://github.com/habla-liaa/encodecmae/tree/v.1.0.0
Still working in updating training scripts for new version (models going from melspectrogram to encodec).

**Comment by DEDSEC-Roger** [comment]
Is there any big changes between the v1 and v2 version pretraining scripts? I currently change the "configs/features/wav_only.gin" to "configs/features/mel.gin" in the run_pretraining.sh, is there any other changes I should do?

**Comment by mrpep** [comment]
That should do it! By the way, checkpoints are available to download for all the paper models. For example, calling load_model('mel256-ec-base'). All available models are here: https://huggingface.co/lpepino/encodecmae-v2/tree/main

**Comment by DEDSEC-Roger** [comment]
In this model "mel256-ec-base-fma.pt", did you just use this data "fma_large.zip: 106,574 tracks of 30s, 161 unbalanced genres (93 GiB)" to train the model?

**Comment by DEDSEC-Roger** [comment]
I am trying my best to reproduce the "mel256-ec-base-fma.pt" model, I just replace "configs/features/wav_only.gin" with "configs/features/mel.gin" and have trained the model for 64.7k step, but the performance is still inferior to your release model, I wonder what have I missed, should I remove the "QUANTIZER_WEIGHTS"? Because I did not find it in "mel256-ec-base-fma.pt"'s config_str.

**Comment by mrpep** [comment]
Yes, it's only with FMA-large. Don't remove quantizer weights. And the problem I think is that the model is trained for 500k steps, it should make a difference training longer in downstream performance

**Comment by DEDSEC-Roger** [comment]
Hey, I am trying to reproduce the self-training stage, I already train a large model using my own data. I wonder what does the st stage loss is, can I reuse the EnCodecMAEClassificationLoss?

**Comment by DEDSEC-Roger** [comment]
Do anyone reproduce the self-training stage successfully? I meet a serious performance degradation after self-training.

**Comment by mrpep** [comment]
Hi, sorry for the delay! Here it's an example of how to run the self-training stage. The first command will generate discrete targets for all the dataset. I found it's important to use the self-attention branch from the ResiDual normalization to train k-means. In the second command it will further train the model with these targets. Differently from version 1, here the original EnCodec targets are discarded, as it gives a slight improvement for some tasks:

```
ginpipe configs/base/create_st_dataset.gin \
		configs/datasets/audioset-unbalanced-24k.gin \
		configs/datasets/fma-large-24k.gin \
		configs/datasets/librilight-6k-24k.gin \
		configs/features/mel.gin \
		--module_list configs/imports \
		--project_name mel256-ec-base_model \
		--experiment_name self_training_dataset_no-pn \
		--mods "TEACHER_MODEL='mel256-ec-base'" \
			"POSTNORM_LAST_ACTIVATION=False" \
			"DEVICE='cuda:0'"

ginpipe configs/base/self-train.gin \
		configs/models/encodecmae.gin \
		configs/features/mel.gin \
		configs/features/st.gin \
		--module_list configs/imports \
		--project_name mel256-ec-base_model \
		--experiment_name st_model-no-pn \
		--mods "ST_CHECKPOINT='huggingface:lpepino/encodecmae-pretrained/upstreams/mel256-ec-base.pt'" \
				"ST_DATASET_DIR='/workspace/encodecmae/encodecmae/experiments/mel256-ec-base_model/self_training_dataset_no-pn'" \
				"DEVICE=[0]"
```

**Comment by mrpep** [comment]
Added the pretraining script in this [commit](https://github.com/habla-liaa/encodecmae/commit/2dd6759c595bf20c9d836914fef5cde9a6386843)
